src/gcp/auth.ts

```diff
@@ -36,3 +86,49 @@ export async function updateAuthDomains(project: string, authDomains: string[]):
   );
   return res.body.authorizedDomains;
 }
+
+/**
+ * getAuthUser
+ * @param project project identifier.
+ * @param email the users email to lookup.
+ * @param phone the users phone number to lookup.
+ * @param uid the users id to lookup.
+ * @param limit the amount of results to return.
+ * @param offset the amount to page through the results
+ * @return an array of user info
+ */
+export async function getAuthUser(
+  project: string,
+  email?: string,
+  phone?: string,
+  uid?: string,
+  limit = 1000,
+  offset = 0,
+): Promise<{ recordCount: string; userInfo: UserInfo }[]> {
+  const expression: { email?: string; phoneNumber?: string; userId?: string } = {};
+  if (email !== undefined) {
```

Comment:
Are these `!== undefined` checks needed? I think we could probably simplify this whole block to:
```
const expressions = { email, phoneNumber: phone, userId: uid };
```